fdb-record-layer-lucene/src/main/java/com/apple/foundationdb/record/lucene/directory/FDBDirectoryLockFactory.java

```diff
@@ -232,6 +235,7 @@ private void fileLockClearFlushAndClose(boolean isRecovery) {
                 agilityContext.asyncToSync(LuceneEvents.Waits.WAIT_LUCENE_FILE_LOCK_CLEAR,
                         agilityContext.apply(fileLockFunc));
             }
+            agilityContext.recordEvent(LuceneEvents.Events.LUCENE_FILE_LOCK_DURATION, System.nanoTime() - lockStartTime);
```

Comment:
If we fail to `flush` and close in a recovery path this would get emitted twice. Perhaps we should only do this if `agilityContext.flush()` succeeds?

Reply:
Discussed with @jjezra and it looks like if we recorded the event once, the `close` state will change to `TRUE` and the next call to `fileLockClearIfLocked` will be noop. We will only enter `fileLockClearFlushAndClose` again in case the `close` failed (and the event was not recorded).